private-templates-service/client/src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { ReactElement } from "react";
+import {
+  Collapse,
+  Container,
+  Navbar,
+  NavbarToggler,
+  NavbarBrand,
+  Nav,
+  NavItem,
+  NavLink,
+  UncontrolledDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from "reactstrap";
+
+// Components
+import { UserType } from '../../App';
+
+// Styles
+import "@fortawesome/fontawesome-free/css/all.css";
+import { AvatarIcon, DefaultAvatarIcon } from './styled';
+
+function UserAvatar(props: { user: UserType }): ReactElement {
+  // If a user avatar is available, return an img tag with the pic
+  if (props.user.avatar) {
+    return (
+      <AvatarIcon
+        src={props.user.avatar}
+        alt="user"
+        className="rounded-circle align-self-center mr-2"
+        style={{ width: "32px" }}
+      ></AvatarIcon>
+    );
+  }
+
+  // No avatar available, return a default icon
+  return (
+    <DefaultAvatarIcon
+      className="far fa-user-circle fa-lg rounded-circle align-self-center mr-2"
+      style={{ width: "32px" }}
+    ></DefaultAvatarIcon>
+  );
+}
+
+function AuthNavItem(props: NavBarProps): ReactElement {
+  // If authenticated, return a dropdown with the user's info and a
+  // sign out button
+  if (props.isAuthenticated) {
+    return (
+      <UncontrolledDropdown>
+        <DropdownToggle nav caret>
+          <UserAvatar user={props.user} />
+        </DropdownToggle>
+        <DropdownMenu right>
+          <h5 className="dropdown-item-text mb-0">
+            {props.user.displayName}
+          </h5>
+          <p className="dropdown-item-text text-muted mb-0">
+            {props.user.email}
+          </p>
+          <DropdownItem divider />
+          <DropdownItem onClick={props.authButtonMethod}>
+            Sign Out
+					</DropdownItem>
+        </DropdownMenu>
+      </UncontrolledDropdown>
+    );
+  }
+
+  // Not authenticated, return a sign in link
+  return (
+    <NavItem>
+      <NavLink onClick={props.authButtonMethod}>Sign In</NavLink>
+    </NavItem>
+  );
+}
+
+export interface NavBarProps {
+  isAuthenticated: boolean,
+  authButtonMethod: () => void,
+  user: UserType,
+}
+
+interface State {
```

Comment:
NavBarState, since otherwise we will have two State interfaces (one from App.tsx and one here)

Reply:
I updated the name.

I am not familiar with convention of how the types of states are defined in TS, though I thought it would be okay to have a `interface State {...}` in many files, as long as they are not exported. If they are exported, then they will be given a more unique name.

Reply:
No you are right, as long as its not exported it is fine.  I was thinking props (in which the interface would be exported).  Resolving